src/graph/flow.rs

```diff
@@ -221,4 +225,53 @@ mod test {
         assert_eq!(cost, 18);
         assert_eq!(flow, 10);
     }
+
+    #[test]
+    fn test_max_matching()
+    {
+        let mut graph = FlowGraph::new(14, 4);
+
+        let source = 0;
+        let sink = 13;
+
+        let a_start = 1;
+        let b_start = 7;
+        //6 nodes in A
+
+        for a in a_start..a_start + 6 {
+            graph.add_edge(source, a, 1, 1);
+        }
+
+        //6 nodes in B
+        for b in b_start..b_start + 6 {
+            graph.add_edge(b, sink, 1, 1);
+        }
+
+        graph.add_edge(a_start + 0, b_start + 1, 1, 1);
+        graph.add_edge(a_start + 0, b_start + 2, 1, 1);
+        graph.add_edge(a_start + 2, b_start + 0, 1, 1);
+        graph.add_edge(a_start + 2, b_start + 3, 1, 1);
+        graph.add_edge(a_start + 3, b_start + 2, 1, 1);
+        graph.add_edge(a_start + 4, b_start + 2, 1, 1);
+        graph.add_edge(a_start + 4, b_start + 3, 1, 1);
+        graph.add_edge(a_start + 5, b_start + 5, 1, 1);
+
+        let (flow_amt, flow) = graph.dinic_flow(source, sink);
+        assert_eq!(flow_amt, 5);
+
+        //U->V edges in maximum matching
+        assert_eq!(
+            flow
```

Comment:
Nice! Consider giving this expression a variable name :P

Reply:
Also consider using into_iter() which consumes flow and lets you use f instead of *f

Reply:
A nice tip, changed.